source/collections/LinkedList.ooc

```diff
@@ -117,13 +116,15 @@ LinkedList: class <T> {
 		}
 		result
 	}
-	removeNode: func (toRemove: Node<T>) {
+	removeNode: func (toRemove: Node<T>) -> T {
 		toRemove prev next = toRemove next
 		toRemove next prev = toRemove prev
 		toRemove prev = null
 		toRemove next = null
+		data := toRemove data
 		toRemove free()
 		this _size -= 1
+		data
```

Comment:
What happens here if you put an object into node and call `removeNode()` ? `Node` calls `free` on the `Cell` object, and `Cell free()` object calls `gc_free` on the data it contains. If you intend to return and keep using the data from `Node`, shouldn't this be more like
```ooc
removeNode: func (toRemove: Node<T>) -> T {
 		toRemove prev next = toRemove next
 		toRemove next prev = toRemove prev
 		toRemove prev = null
 		toRemove next = null
		data := toRemove data
		toRemove data = null  // <-- no `free` will be called on `data` now
 		toRemove free()
 		this _size -= 1
		data
}
```
?